File: src/lib/pjecalc/core/dominio/calculo/relatorio/salario-familia-jr-adapter.ts
/**
 * PJe-Calc v2.15.1 — SalarioFamiliaJRAdapter (abstract)
 * Porte TS-adaptado de:
 *   br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio.SalarioFamiliaJRAdapter
 *
 * Ref Java: pjecalc-fonte/.../relatorio/SalarioFamiliaJRAdapter.java
 */
import type Decimal from 'decimal.js';
import { JRAdapter, JRAdapterDataSource } from './jr-adapter';

export abstract class SalarioFamiliaJRAdapter extends JRAdapter {
  abstract getOcorrencias(): JRAdapterDataSource<SalarioFamiliaOcorrenciaAdapter>;
  abstract getQuantidadeDependentes(): number;
  abstract getTotalGeral(): Decimal;
}

export abstract class SalarioFamiliaOcorrenciaAdapter extends JRAdapter {
  abstract getCompetencia(): Date;
  abstract getBase(): Decimal;
  abstract getValorPorCota(): Decimal;
  abstract getQuantidade(): number;
  abstract getValorDevido(): Decimal;
  abstract getIndiceCorrecao(): Decimal;
  abstract getValorCorrigido(): Decimal;
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/relatorio/SalarioFamiliaJRAdapter.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.relatorio;

import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapter;
import br.jus.trt8.pjecalc.base.comum.relatorio.JRAdapterDataSource;
import java.math.BigDecimal;
import java.util.Date;

public abstract class SalarioFamiliaJRAdapter
extends JRAdapter {
    public abstract String getFormula();

    public abstract String getPeriodo();

    public abstract JRAdapterDataSource<OcorrenciaSalarioFamiliaAdapter> getOcorrencias();

    public abstract BigDecimal getTotalDoDevidoCorrigido();

    public abstract BigDecimal getTotalDeJuros();

    public abstract BigDecimal getTotalGeral();

    public abstract class OcorrenciaSalarioFamiliaAdapter
    extends JRAdapter {
        public abstract Date getOcorrencia();

        public abstract BigDecimal getSalarioReferencia();

        public abstract BigDecimal getSalarioParaFaixa();

        public abstract Integer getQuantidadeFilhos();

        public abstract BigDecimal getDevido();

        public abstract BigDecimal getIndice();

        public abstract BigDecimal getValorCorrigido();

        public abstract BigDecimal getJuros();

        public abstract BigDecimal getTotal();
    }
}

